src/compiler/checker.ts

```diff
@@ -8965,10 +8966,13 @@ namespace ts {
             if (left.kind === SyntaxKind.SuperKeyword) {
                 return true;
             }
-            // A protected property is accessible in the declaring class and classes derived from it
+            // A protected property is accessible if the property is within the declaring class or classes derived from it
+            const typeClassDeclaration = <ClassLikeDeclaration>getClassLikeDeclarationOfSymbol(type.symbol);
             if (!enclosingClass || !hasBaseType(enclosingClass, declaringClass)) {
-                error(node, Diagnostics.Property_0_is_protected_and_only_accessible_within_class_1_and_its_subclasses, symbolToString(prop), typeToString(declaringClass));
-                return false;
+                if (!isNodeWithinClass(node, typeClassDeclaration)) {

```

Comment:
Ah, yes, I see, my bad.

One last comment. Should I rewrite `isNodeWithinClass` to be `Type` oriented instead of `ClassLikeDeclaration` oriented?

`hasBaseType()` takes `InterfaceType`, and most other places that call `isNodeWithinClass` gets the `Type`'s `ClassLikeDeclaration` _(which could be skipped if it was `Type` oriented)_.
If we make 
`isNodeWithinClass(node: Node, classDeclaration: ClassLikeDeclaration)` into `isNodeWithinType(node: Node, type: Type)` _(or maybe a better name)_,
and convert the `getContainingClass` to a `Type` within `forEachEnclosingClass<T>()`

Or should we keep it as-is and convert the `ClassLikeDeclaration` to a `Type` within the callback of  `forEachEnclosingClass<T>()`?